sympy/functions/special/delta_functions.py

```diff
@@ -218,12 +234,25 @@ def eval(cls, arg):
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero and H0 != S.NaN:
+            return H0
```

Comment:
Nope.  My bad.  `None` raises the exception earlier, before `eval()` is reached in the `Function` class.  But the consequence is the same -- `None` doesn't work well as the value to pass for restoring the default behavior.

But passing in a `NaN` for the second argument will work.